src/ClientHandle.h

```diff
@@ -408,6 +410,13 @@ class cClientHandle  // tolua_export
 
 	cPlayer * m_Player;
 
+	/** This is an optimization which saves you an iteration of m_SentChunks if you just want to know
+	whether or not the player is standing at a sent chunk.
+	If this is equal to the coordinates of the chunk the player is currrently standing at, then this must be a sent chunk
+	and a member of m_SentChunks.
+	Otherwise, this contains an arbitrary value which should not be used. */
+	cChunkCoords m_CachedSentChunk;
```

Comment:
Are you sure? This is not just a micro optimization. Without this, an extra lock will be held per player per tick, and every tick, `m_SentChunks` would be fully iterated per player.

If you don't like the extra variable, we could turn `m_SentChunks` into an unordered map and get rid of this. I need an O(1) answer to the question of `is this chunk sent?` and an unordered map provides that.